sql/analyzer/resolve_tables.go

```diff
@@ -43,8 +43,17 @@ func resolveTables(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel
 			// lower in the tree.
 			var resolvedTables []sql.Node
 			for _, t := range p.Children() {
-				if _, ok := t.(*plan.ResolvedTable); ok {
+				switch tbl := t.(type) {
+				case *plan.ResolvedTable:
 					resolvedTables = append(resolvedTables, t)
+				case *plan.UnresolvedTable:
+					continue
+				case *plan.SubqueryAlias:
+					// the views are resolved to be SubqueryAlias node
+					return p, transform.SameTree, sql.ErrUnknownTable.New(tbl.Name())
+				default:
+					// TODO: should other nodes be skipped as well?
+					continue
```

Comment:
I'm wondering if there is a cleaner way to handle this... 

It looks like we special case `DropTable` here in `resolveTables` in order to support the `drop table if exists ...` syntax so that we can cleanly execute `drop table` statements that refer to tables that don't resolve. 

Another way to handle this might be to remove this special handling for `DropTable` from `resolveTables` and just let resolution happen normally. Then we already have a `validateDropTable` analyzer rule that we could update to validate that every child of `DropTable` is a resolved table that can be deleted, and not a view. If the `if exists` clause is specified, then we could simply skip any unresolved tables. The only catch there is that `validateResolved` is going to now need a special case for `DropTable`, because it's valid for it to include tables that don't actually exist. 

This feels slightly cleaner to me (e.g. `DropTable` validation done in the `validateDropTable` function/rule instead of peppered through out other rules), and it has the advantage that it would enable us to add a warning to the SQL shell for non-existent tables, just like MySQL does, instead of silently dropping them like we do now. 

Think about that and let me know what you think. I think it ends up being overall slightly cleaner that way, but I'm still thinking through the details.